### ISSUE TITLE
Error on Task Azure File Copy: "The requested security protocol is not supported"

### ISSUE THREAD

**Issue by fajanssen** [Microsoft/vsts-tasks#6895]
When trying to run Task Azure File Copy to ARM Storage Account in a release I get the following Error:
##[error]Failed to add TLS 1.2 in current session. Error: 'Exception setting "SecurityProtocol": "The requested security protocol is not supported."'

**Comment by iratherscribble** [comment]
Also getting the issue. Seems to be something that was introduced in version 1.0.111 of the VSTS task. In version 1.0.110 of the task, it was executing without the error.

Even though it is being reported as an error in VSTS, it isn't causing the task to fail and appears the files/blobs get copied. So it purely seems to be a visual issue. 

Wondering if the task should be logging it as an error at all since the copy is successful and there is nothing the user can/or needs to do to resolve the error.

**Comment by rajatagrawal-dev** [comment]
Hi @fajanssen, @iratherscribble 
For a few tasks, we have begun adding TLS 1.2 in powershell sessions in which the task runs on the agent. If you are getting the error mentioned, it means that the task failed to add TLS 1.2 in session. Even though your tasks might be passing as of now, they may fail in the near future as TLS 1.2 becomes a mandatory requirement. Hence, it is shown as an error. But it can be considered as a strong warning right now.

Can you please share the OS details and powershell version of your agent? I'll see if I can repro the issue of why adding TLS 1.2 in session failed.

**Comment by fajanssen** [comment]
hi rajatagrawal-dev,

The Agents Powershell version is 3.6.0 and it's a hosted VS2017 Windows Agent

**Comment by fajanssen** [comment]
I can confirm that the Task is executed although it shows an error and files get copied.

**Comment by rajatagrawal-dev** [comment]
@Farzad-Jalali The fix might take a couple of days to reach all accounts. But as stated above, TLS error should not cause any blocking failure. Let me know if that isn't the case for you.

**Comment by ermontgo** [comment]
I am seeing what looks like a related error in the DACPAC task. When it executes, we get the following error at the start of the task:

[error]Failed to add TLS 1.2 in current session. Error: 'Exception setting "SecurityProtocol": "The requested security protocol is not supported."'

The task then continues, but fails to connect with the following message:
Execute-Command : Unable to connect to master or target server '(Database Name)'. You must have a user with the same password in master or target server '(Database Name)'. 

We verified that the password is correct and logs in correctly through SSMS, and that the SqlPackage command has the right values.

Could this be the same issue?

**Comment by rajatagrawal-dev** [comment]
@ermontgo The reason your task is failing is **not** related to TLS. 
I suspect there might be an issue with how you are passing the parameters to the task or some firewall issue. Please raise a new issue along with your debug logs to help us unblock you.

**Comment by rajatagrawal-dev** [comment]
@fajanssen @iratherscribble @hoffmanic @Farzad-Jalali @ermontgo 
Please check if you are still facing the issue. If yes, then please share your account name, the task in which you are seeing the error, and the version of the task your releases are picking.

**Comment by Teddy1313** [comment]
When was the fix deployed to all VSTS accounts?

**Comment by rajatagrawal-dev** [comment]
@Teddy1313 Fix was deployed a couple of days ago. If you are still facing the issue then please share your account name, the task in which you are seeing the error, and the version of the task your releases are picking.

**Comment by rajatagrawal-dev** [comment]
Closing this as issue is now resolved.

**Comment by PradeepKumarMunirathinam** [comment]
Hi , 

I am using Azure File copy task in VSTS to copy a folder from local git into Azure blob storage.
The above task is working fine in Build.
When try the same in Release, The task fails with the below error
 **failed with error: '' For more info please refer to https://aka.ms/azurefilecopyreadme**

I tried the Release with HOSTED Agent and Local Agent. And both throws the same error .

Local Agent details : 
Agent.OS = Windows_NT
Agent.OSVerison = 6.3.9600
Agent.Version = 2.124.0
AzurePS = 5.0.1


Kindly revert if there is any fix to it.

**Comment by rajatagrawal-dev** [comment]
@PradeepKumarMunirathinam Can you please open a new issue with the details. This issue is related to TLS protocol issue and is now closed. New issue will help get the right traction.

Please also include the debug logs when you create the new issue. You can get the debug logs by adding a variable _system.debug_ in your release definition and setting its value to _true_. You can send the logs to RM_Customer_Queries [at] microsoft [dot] com, in case you don't want to post them here.
